joplin/js/editor.js

```diff
@@ -203,46 +169,43 @@ $(function() {
     let labelList = state.languageLabels;
-    for (let label in labelList) {
-      if (labelList[label].querySelector) {
-        let languageTag = labelList[label].querySelector('span').innerText;
-        // these seem to be nested twice, from the title to the containing element
-        // TODO: come up with a more elegant and maintable way to check what elements ought to be hidden
+    for (let label of labelList) {
+      if (label.querySelector) {
+        let languageTag = label.querySelector('span').innerText;
+        // TODO: come up with a more elegant and maintainable way to check what elements ought to be hidden
         if (
-          labelList[label].parentElement.parentElement.parentElement.classList
-            .value !== 'struct-block'
+          label.parentElement.parentElement.parentElement.classList
+            .value !== 'rah-static rah-static--height-auto c-sf-block__content'
         ) {
-          const translatedElement =
-            labelList[label].parentElement.parentElement;
-
+          const translatedElement = label.parentElement.parentElement;
           if (languageTag != null && languageTag != currentLang) {
             translatedElement.classList.add('hidden');
           } else {
             translatedElement.classList.remove('hidden');
           }
-          /*  While the first condition checks for 'struct-blocks' with language tags,
+          /*  check if still valid with new react streamfield
+              While the first condition checks for 'struct-blocks' with language tags,
               it doesn't catch the case where there are 'struct-blocks' with language
-              tages WITHIN the element itself. The following condition checks for those conditions.
-              - The was neccessary for guide stream fields. */
-          if (translatedElement.classList.contains('struct-block')) {
-            const fieldlabels = translatedElement.querySelectorAll('[for]');
-            fieldlabels.forEach(fieldlabel => {
-              const attrFor = fieldlabel.getAttribute('for').split('_');
-              fieldlabel.parentNode.classList.remove('hidden');
-              // Adding a failsafe to make sure we don't remove non translated fields
-              const attrLang = attrFor[attrFor.length - 1];
-              if (['en', 'es', 'vi', 'ar'].includes(attrLang)) {
-                if (attrLang !== currentLang) {
-                  fieldlabel.parentNode.classList.add('hidden');
-                  translatedElement.classList.remove('hidden'); // only re-reveal the parent class if we find this case.
-                }
-              }
-            });
-          }
+              tags WITHIN the element itself. The following condition checks for those conditions.
+              - The was necessary for guide stream fields. */
+//          if (translatedElement.classList.contains('struct-block')) {
+//            const fieldlabels = translatedElement.querySelectorAll('[for]');
```

Comment:
I don't think this is needed right now, but I wanted to just comment it out instead of removing it in case we a)actually do need it and b) it is needed soon

Reply:
I believe this is part of the way we switch out fields based on the selected language?

Reply:
its for a specific case of nested streamfields that doesnt seem to be happening in the current implementation.